src/main/scala/it/pagopa/pdnd/interop/uservice/partymanagement/model/persistence/commands.scala

```diff
@@ -36,20 +37,33 @@ final case class ConfirmPartyRelationship(
   relationshipId: UUID,
   filePath: String,
   fileInfo: FileInfo,
+  timestamp: OffsetDateTime,
   replyTo: ActorRef[StatusReply[Unit]]
 ) extends PartyRelationshipCommand
 
-final case class RejectPartyRelationship(relationshipId: UUID, replyTo: ActorRef[StatusReply[Unit]])
-    extends PartyRelationshipCommand
+final case class RejectPartyRelationship(
+  relationshipId: UUID,
+  timestamp: OffsetDateTime,
+  replyTo: ActorRef[StatusReply[Unit]]
+) extends PartyRelationshipCommand
 
-final case class SuspendPartyRelationship(relationshipId: UUID, replyTo: ActorRef[StatusReply[Unit]])
-    extends PartyRelationshipCommand
+final case class SuspendPartyRelationship(
+  relationshipId: UUID,
+  timestamp: OffsetDateTime,
```

Comment:
why is the timestamp modeled in the command itself? Wouldn't be enough to pass it to events only, i.e.: generating the timestamp at PartyPersistentBehavior when we are going to persist the event?

Probably commands should model what an operation does, not when.

Reply:
It's not actually a `when`, it's a `what` with a timestamp type.

I was thinking the same, but I did not put it into the behavior because:
- this way commands are always fully reproducible
- generating timestamps in the behavior means that if (when) events are reprocessed, all timestamps will be updated at `now`

Reply:
AFAIK, in event sourcing are the Events that needs to be repeatable, while Commands should not, since they may have side effect too.

Related to your second consideration, namely:

> generating timestamps in the behavior means that if (when) events are reprocessed, all timestamps will be updated at now

I don't see the problem of doing something like:

`Event.persist(WhateverGenerated(payload, OffsetDateTime.now()))` since the event persistence is persisting the OffsetDateTime.now as well. So, any snapshot restore or event replication would not generate another timestamp at all.  Bonus points: 

 - command model will be leaner;
 - timestamp events management would be encapsulated in the Persistent Actor class only, since they won't leak downstream from the Service class to command and later to events.

Indeed, even Akka itself adds a timestamp to EventEnvelope by default:

https://doc.akka.io/japi/akka/current/akka/persistence/query/EventEnvelope.html

Moreover, in terms of pure event sourcing ortodoxy we may implicitly assign a semantic to event timestamp given event model meaning. E.g.: PartyCreated event with a timestamp should implicitly state on the global events _ledger_ that the event timestamp represents the creation time of the party (i.e.: createdAt), since that event type represents a creation command occurred.

However, since we are not implementing a pure CQRS, please pick the solution you think fits the most.

Reply:
no, you are right, I was making the wrong assumptions.
I'll move the timestamp generation to the behavior.

It's important to find the right solution because we are going to add these timestamps everywhere.

Thanks for pointing it out!